src/connector.rs

```diff
@@ -285,8 +291,12 @@ impl EthConnectorContract {
             // Store proof only after `mint` calculations
             self.record_proof(&data.proof_key);
             // Save new contract data
-            self.save_contract();
+            self.save_ft_contract();
 
+            // TODO: Why return a promise here instead of just call it here directly?
```

Comment:
If I remember correctly this was made to follow the architecture by creating a transfer to Aurora via `ft_transfer_call`. If we don't create a promise and just make a regular function call, we won't be able to verify that this was called from the Aurora contract.